app/controllers/api/databases.php

```diff
@@ -1673,12 +1676,77 @@ function updateAttribute(
             throw new Exception(Exception::COLLECTION_NOT_FOUND);
         }
 
-        $attributes = $collection->getAttribute('attributes');
+        $queriesFromRequest = Query::parseQueries($queries);
 
-        $response->dynamic(new Document([
-            'total' => \count($attributes),
-            'attributes' => $attributes
+        // Add type property in query if select query exists and type property doesn't exist as type is required for response model
+        $hasDetailsinQuery = [
+            'selectQuery' => false,
+            'typeAttributeInSelectQuery' => false,
+        ];
+        foreach ($queriesFromRequest as $query) {
+            if ($query->getMethod() ===  Query::TYPE_SELECT) {
+                $hasDetailsinQuery['selectQuery'] = true;
+            }
+            if (\array_search('type', $query->getValues())) {
+                $hasDetailsinQuery['typeAttributeInSelectQuery'] = true;
+            }
+        }
+        $transformedQueries = $queriesFromRequest;
+        if ($hasDetailsinQuery['selectQuery'] && !$hasDetailsinQuery['typeAttributeInSelectQuery']) {
+            \array_push($transformedQueries, Query::select(['type']));
+        }
+
+        \array_push($transformedQueries, Query::equal('collectionId', [$collectionId]), Query::equal('databaseId', [$databaseId]));
+
+        // Get cursor document if there was a cursor query
+        $cursor = Query::getByType($transformedQueries, [Query::TYPE_CURSORAFTER, Query::TYPE_CURSORBEFORE]);
+        $cursor = reset($cursor);
+
+        if ($cursor) {
+            $attributeId = $cursor->getValue();
+            $cursorDocument = Authorization::skip(fn() => $dbForProject->find('attributes', [
+                Query::equal('collectionId', [$collectionId]),
+                Query::equal('databaseId', [$databaseId]),
+                Query::equal('key', [$attributeId]),
+                Query::limit(1),
+            ]));
+            if (empty($cursorDocument) || $cursorDocument[0]->isEmpty()) {
+                throw new Exception(Exception::GENERAL_CURSOR_NOT_FOUND, "Attribute '{$attributeId}' for the 'cursor' value not found.");
+            }
+            $cursor->setValue($cursorDocument[0]);
+        }
+
+        $attributes = $dbForProject->find('attributes', $transformedQueries);
+        $filterQueries = Query::groupByType($transformedQueries)['filters'];
+        $total = $dbForProject->count('attributes', $filterQueries, APP_LIMIT_COUNT);
+
+         //Add relationship data from options to attributes as it loses options during response setup
+        foreach ($attributes as $attribute) {
+            if ($attribute->getAttribute('type') === Database::VAR_RELATIONSHIP) {
```

Comment:
Do we need the `type` for the response models or just to check if it's a relationship here?

Reply:
This logic adds relationship attributes to each attribute object as the options key gets filtered out from Attribute Relationship Type as defined in the model. 
I have moved this logic in the filter function of the Relationship Model so we can avoid this from controller logic.

Reply:
If we left it in the controller, would we be able to remove the type attribute after fetching from the database but before we call `$response->dynamic`? It would be much cleaner that way as opposed to calling `$response->output` then `$response->static` after filtering

Reply:
This part of code is not really related to do with $type attribute. 

This is just to move from options key which contains relationship data to the appropriate keys that are required according to the models.
Regarding output and static 
That is not really possible, even I didn't like doing it but the whole reason of adding that type in select query is required because output method requires the type property. 

We cannot remove type property before output is run